### PR TITLE
Add cleanup command which is executed at the end of each benchmark

### LINKED ISSUE #91: Provide preparation command to be run between benchmarking different commands
I want to benchmark two versions of `rustc` against each other, measuring the build time of a project that however has some build dependencies, downloading and building which I do not want included in the benchmark.  However, after using one version of rustc, I need to `rm -rf target` to enable the other one to build; these are both self-compiled git versions of Rust and their detection of when to recompile because the compiler changed does not work reliably.

Unfortunately, this is not possible with one hyperfine invocation currently; instead I have to resort to calling hyperfine twice and clean the build dir in between. It would be great if hyperfine had a preparation command that was run in-between different commands being benchmarked, but NOT in-between different runs of the same benchmark.

### PR DESCRIPTION
**Pull request by colinwahl**
Fixes #91. 

I am still a beginner at Rust, so I'm very open to feedback. Thanks!

### CODE REVIEW

**Comment by sharkdp** [comment]
Thank you very much for your contribution!

I will do a full review soon, but what I've seen so far looks great. I think we should work a little bit on the `--help` text to clearly explain what the difference from `--prepare` is. The prepare-command will be executed before *each* benchmarking run (several times for every benchmarked command) while the cleanup command will only be called once after a set of benchmark runs (once for every benchmarked command).

File: src/hyperfine/app.rs [hunk 1]
```diff
@@ -82,6 +82,17 @@ fn build_app() -> App<'static, 'static> {
...
                 ),
         )
+        .arg(
+            Arg::with_name("cleanup")
+                .long("cleanup")
+                .short("c")
+                .takes_value(true)
+                .value_name("CMD")
+                .help(
+                    "Execute CMD after each benchmark run. This is useful for \
```
Comment (sharkdp) [inline src/hyperfine/app.rs]: As stated previously, we should clarify the help text here.

Comment (colinwahl) [reply src/hyperfine/app.rs]: I agree it needs some clarification, how does this sound?

"Execute CMD after the benchmarking runs for each individual command to be benchmarked.  This is useful if the commands to be benchmarked produce artifacts that need to be cleaned up."

Comment (sharkdp) [reply src/hyperfine/app.rs]: Sounds great!

Comment (colinwahl) [reply src/hyperfine/app.rs]: Ok! I have updated the code.

**Pull request merged**